### PR TITLE
GDB-10937: Reloads similarity indexes and ChatGPT connectors when the repository is changed

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
 - When the repository in the create/edit agent form is changed, the options in "Full-text search" and "Similarity search" are incorrect.

## Why
 - The options are generated when the dialog is opened and are based on the selected repository in the workbench.

## How
 - Implemented logic in the create/edit agent to refresh the options and selected values for "Full-text search," "Similarity search," and "ChatGPT retrieval connector" when the repository is changed;
 - Changed the functionality for expanding and collapsing query methods;
 - Modified the behavior of the authentication interceptor that reads the saved repository ID and location from local storage. Implemented a check to determine if the request headers contain repository headers. If not, the interceptor retrieves them from local storage; otherwise, it uses the headers passed in the request;
 - Extended the similarity service to include two optional parameters: repository ID and repository location;
 - Updated the similarity options to refresh when the panel is opened or when the repository is changed;
 - Extended the connectors service to include two optional parameters: repository ID and repository location;
 - Updated the ChatGPT options to refresh when the panel is opened or when the repository is changed.

### CODE REVIEW

File: src/js/angular/core/services/connectors.service.js [hunk 1]
```diff
@@ -13,44 +13,62 @@ ConnectorsService.$inject = ['ConnectorsRestService'];
...
+     * Fetches all connector types from the server for the repository with id <code>repositoryId</code> and location <code>repositoryLocation</code>.
+     * If the repository ID and repository location are not provided, the values persisted in local storage will be used {@see authentication.interceptor.js}.
+     *
+     * @param {string} repositoryId - (optional) The repository id.
+     * @param {string} repositoryLocation - (optional) The repository location.
      * @return {Promise<ConnectorTypesListModel>}
      */
-    const getConnectorTypes = () => {
-        return ConnectorsRestService.getConnectors()
+    const getConnectorTypes = (repositoryId, repositoryLocation) => {
```
Comment (yordanalexandrov) [inline src/js/angular/core/services/connectors.service.js:23]: I don't like it.

You are describing logic that is happening somewhere else and can be changed.
Maybe remove this from the comment and only state that if user passes id and location, they will be used explicitly

File: src/js/angular/ttyg/controllers/agent-settings-modal.controller.js [hunk 2]
```diff
@@ -200,24 +195,70 @@ function AgentSettingsModalController(
...
         $uibModalInstance.dismiss({});
     };
 
+    /**
+     * Updates the similarity search panel.
+     *
+     * @param {boolean} clearIndexSelection - If true, the selected index will be cleared.
+     */
+    $scope.updateSimilaritySearchPanel = (clearIndexSelection = false) => {
+        const similaritySearchExtractionMethod = $scope.agentFormModel.assistantExtractionMethods.extractionMethods
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/controllers/agent-settings-modal.controller.js:215]: Move this `find` in the model please

File: src/js/angular/ttyg/controllers/agent-settings-modal.controller.js [hunk 3]
```diff
@@ -200,24 +195,70 @@ function AgentSettingsModalController(
...
+    };
+
+    /**
+     * Updates the ChatGPT retrieval connector panel.
+     *
+     * @param {boolean} clearSelection - If true, the selected connector will be cleared.
+     */
+    $scope.updateRetrievalConnectorPanel = (clearSelection = false) => {
+        const retrievalExtractionExtractionMethod = $scope.agentFormModel.assistantExtractionMethods.extractionMethods
+            .find((extractionMethod) => extractionMethod.method === ExtractionMethod.RETRIEVAL);
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/controllers/agent-settings-modal.controller.js:230]: same as above

**Review by yordanalexandrov — APPROVED** [review]

File: src/js/angular/ttyg/controllers/agent-settings-modal.controller.js (hunk 2 above)
Comment (boyan-tonchev) [reply src/js/angular/ttyg/controllers/agent-settings-modal.controller.js]: done

File: src/js/angular/ttyg/controllers/agent-settings-modal.controller.js (hunk 3 above)
Comment (boyan-tonchev) [reply src/js/angular/ttyg/controllers/agent-settings-modal.controller.js]: Done

**Comment by boyan-tonchev** [comment]
Jenkins retry build

File: src/js/angular/core/services/connectors.service.js [hunk 4]
```diff
@@ -13,44 +13,61 @@ ConnectorsService.$inject = ['ConnectorsRestService'];
...
     };
 
     /**
-     * Fetches the connector prefix by its name.
+     * Fetches the connector prefix by its name for the repository with id <code>repositoryId</code> and location <code>repositoryLocation</code>.
+     * If the repository ID and repository location are not provided, the currently selected repository will be used.
+     *
      * @param {string} name
+     * @param {string} repositoryId - (optional) The repository id.
+     * @param {string} repositoryLocation - (optional) The repository location.
```
Comment (svilenvelikov) [inline src/js/angular/core/services/connectors.service.js:34]: What does it mean `(optional)` for these 2 parameters? Shouldn't optional parameters have default values?

File: src/js/angular/ttyg/templates/modal/agent-settings-modal.html [hunk 5]
```diff
@@ -111,8 +111,14 @@ <h4 class="modal-title">{{(
 
                         <!-- FTS method settings -->
 
-                        <div ng-if="extractionMethod.method === extractionMethods.FTS_SEARCH"
+                        <div ng-if="extractionMethod.expanded && extractionMethod.method === extractionMethods.FTS_SEARCH"
                              class="extraction-method-options">
+                            <button class="btn btn-link btn-sm regenerate-question-btn pull-right"
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/templates/modal/agent-settings-modal.html:116]: this is not a regenerate-question-btn

File: src/js/angular/ttyg/templates/modal/agent-settings-modal.html [hunk 6]
```diff
@@ -135,9 +141,15 @@ <h4 class="modal-title">{{(
 
                         <!-- Similarity method settings -->
 
-                        <div ng-if="extractionMethod.method === extractionMethods.SIMILARITY"
+                        <div ng-if="extractionMethod.expanded && extractionMethod.method === extractionMethods.SIMILARITY"
                              ng-class="{'has-error': extractionMethod.selected && !extractionMethod.sparqlOption}"
                              class="extraction-method-options">
+                            <button class="btn btn-link btn-sm regenerate-question-btn pull-right"
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/templates/modal/agent-settings-modal.html:147]: same as above

File: src/js/angular/ttyg/templates/modal/agent-settings-modal.html [hunk 7]
```diff
@@ -111,8 +111,14 @@ <h4 class="modal-title">{{(
 
                         <!-- FTS method settings -->
 
-                        <div ng-if="extractionMethod.method === extractionMethods.FTS_SEARCH"
+                        <div ng-if="extractionMethod.expanded && extractionMethod.method === extractionMethods.FTS_SEARCH"
                              class="extraction-method-options">
+                            <button class="btn btn-link btn-sm regenerate-question-btn pull-right"
+                                    ng-click="checkIfFTSEnabled()"
+                                    ng-disabled="disabled"
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/templates/modal/agent-settings-modal.html:118]: Where is this `disabled` flag?

File: src/js/angular/core/services/connectors.service.js (hunk 4 above)
Comment (boyan-tonchev) [reply src/js/angular/core/services/connectors.service.js]: Done

File: src/js/angular/ttyg/templates/modal/agent-settings-modal.html (hunk 5 above)
Comment (boyan-tonchev) [reply src/js/angular/ttyg/templates/modal/agent-settings-modal.html]: Removed

File: src/js/angular/ttyg/templates/modal/agent-settings-modal.html (hunk 6 above)
Comment (boyan-tonchev) [reply src/js/angular/ttyg/templates/modal/agent-settings-modal.html]: Removed

File: src/js/angular/ttyg/templates/modal/agent-settings-modal.html (hunk 7 above)
Comment (boyan-tonchev) [reply src/js/angular/ttyg/templates/modal/agent-settings-modal.html]: Removed

**Comment by boyan-tonchev** [comment]
Jenkins retry build

**Comment by boyan-tonchev** [comment]
Jenkins retry build

File: src/css/ttyg/agent-settings-modal.css [hunk 8]
```diff
@@ -52,7 +47,7 @@
     width: 100%
 }
 
-.agent-settings-form .additional-extraction-methods .agent-settings-form [type="range"] {
+.agent-settings-form [type="range"] {
```
Comment (svilenvelikov) [inline src/css/ttyg/agent-settings-modal.css]: Not all range fields were designed this way. Check if you didn't brake something with such broad selector!

Comment (boyan-tonchev) [reply src/css/ttyg/agent-settings-modal.css]: We have three range inputs: 'Similarity Threshold', 'Top_P', and 'Temperature'. All of them should have this styling, but all of them have an ID, so I can use it to create a more specific selector.

Comment (boyan-tonchev) [reply src/css/ttyg/agent-settings-modal.css]: I saw that you fixed this in your MR, so I reverted the changes.

**Comment by svilenvelikov** [comment]
Jenkins retry build

**Pull request merged**